internal/sorter/sorter.go

```diff
@@ -176,10 +176,15 @@ type filterChainSorter []*envoy_api_v2_listener.FilterChain
 func (s filterChainSorter) Len() int      { return len(s) }
 func (s filterChainSorter) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s filterChainSorter) Less(i, j int) bool {
-	// The ServerNames field will only ever have a single entry
-	// in our FilterChain config, so it's okay to only sort
-	// on the first slice entry.
-	return s[i].FilterChainMatch.ServerNames[0] < s[j].FilterChainMatch.ServerNames[0]
+
+	// (SAS) Default
+	if len(s[i].FilterChainMatch.ServerNames) > 0 && len(s[j].FilterChainMatch.ServerNames) > 0 {
+		// The ServerNames field will only ever have a single entry
+		// in our FilterChain config, so it's okay to only sort
+		// on the first slice entry.
+		return s[i].FilterChainMatch.ServerNames[0] < s[j].FilterChainMatch.ServerNames[0]
+	}
+	return false
```

Comment:
I think that this needs to sort filter chains with no ServerNames to the end? We should be able to add a test for that case too.